crates/formats/src/entryfilelist.rs

```diff
@@ -0,0 +1,258 @@
+    pub unk2_count: usize,
+    pub _unkc: u32,
+}
+
+impl<'a> EntryFileList<'a> {
+    pub fn from_bytes(bytes: &'a [u8]) -> Result<Self, EntryFileListError> {
+        let (container_header, compressed) = Ref::<_, ContainerHeader>::new_from_prefix(bytes)
+            .ok_or(EntryFileListError::UnalignedValue)?;
+
+        Ok(Self {
+            container_header: container_header.into_ref(),
+            compressed,
+        })
+    }
+
+    pub fn content(&self) -> Result<SectionIter<'a, Unk1>, EntryFileListError> {
+        let mut decoder = ZlibDecoder::new(self.compressed);
+
+        let _unk0 = decoder.read_u32::<LE>()?;
+        let unk1_count = decoder.read_u32::<LE>()? as usize;
+        let unk2_count = decoder.read_u32::<LE>()? as usize;
+        let _unkc = decoder.read_u32::<LE>()?;
+
+        let header = EntryFileListHeader {
+            _unk0,
+            unk1_count,
+            unk2_count,
+            _unkc,
+        };
+
+        Ok(SectionIter {
+            decoder,
+            entry_count: header.unk1_count,
+            header,
+            entries_read: 0,
+            _marker: PhantomData,
+        })
+    }
+}
+
+impl<'a> std::fmt::Debug for EntryFileList<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("EntryFileList")
+            .field("unk04", &self.container_header._unk04.get())
+            .field(
+                "compressed_size",
+                &self.container_header.compressed_size.get(),
+            )
+            .field(
+                "decompressed_size",
+                &self.container_header.decompressed_size.get(),
+            )
+            .finish()
+    }
+}
+
+#[derive(Debug)]
+pub struct SectionIter<'a, TElement> {
+    decoder: ZlibDecoder<&'a [u8]>,
+    header: EntryFileListHeader,
+    entry_count: usize,
+    entries_read: usize,
+    _marker: PhantomData<TElement>,
+}
+
+impl<'a, TElement> SectionIter<'a, TElement> {
+    fn skip_to_end(&mut self) -> io::Result<()> {
+        if self.entries_read != self.entry_count {
+            let remaining =
+                (self.entry_count - self.entries_read) * std::mem::size_of::<TElement>();
+            std::io::copy(
+                &mut self.decoder.by_ref().take(remaining as u64),
+                &mut std::io::sink(),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    fn skip_to_alignment(&mut self) -> io::Result<()> {
+        // Find distance to next alignment point if we're not already aligned
+        let to_read = {
+            let offset = self.decoder.total_out() as usize % 0x10;
+            if offset == 0 {
+                0
+            } else {
+                0x10 - offset
+            }
+        };
+
+        self.decoder.read_padding(to_read)
+    }
+}
+
+pub trait SectionElement: Sized {
+    fn read(reader: &mut impl Read) -> std::io::Result<Self>
+    where
+        Self: Sized;
+}
+
+impl<'a, TElement> Iterator for SectionIter<'a, TElement>
```

Comment:
`T` is fine here, I think most people would figure out the `T` is the element produced by the iterator.

Reply:
Fixed per https://github.com/soulsmods/fstools-rs/pull/31/commits/99296a8272bd83e90e78bb5d0ae0d8c205146714